### PR TITLE
fix app rating on iOS 8

### PR DESCRIPTION
**Pull request by coreyfloyd**
we need to avoid the internal browser for the app store url

https://phabricator.wikimedia.org/T126839

### CODE REVIEW

File: Wikipedia/Code/WMFSettingsViewController.m [hunk 1]
```diff
@@ -158,7 +158,7 @@ - (void)tableView:(UITableView*)tableView didSelectRowAtIndexPath:(NSIndexPath*)
             [self wmf_openExternalUrl:[NSURL URLWithString:WMFSettingsURLZeroFAQ]];
             break;
         case WMFSettingsMenuItemType_RateApp:
-            [self wmf_openExternalUrl:[NSURL URLWithString:WMFSettingsURLRate]];
+            [[UIApplication sharedApplication] openURL:[NSURL URLWithString:WMFSettingsURLRate]]; //always exit the app
```
Comment (bgerstle) [inline Wikipedia/Code/WMFSettingsViewController.m]: don't we need to have in-between logic for wikipedia zero?

Comment (bgerstle) [inline Wikipedia/Code/WMFSettingsViewController.m]: sorry but, technically yes we do. so i would recommend modifying `wmf_openExternalURL` by adding another `BOOL` parameter which tells the logic whether it should open in app or Safari

Comment (bgerstle) [inline Wikipedia/Code/WMFSettingsViewController.m]: e.g. 

``` objc
- (void)wmf_openExternalURL:(NSURL*)url {
  [self wmf_openExternalURL:url inSafari:NO];
}

- (void)wmf_openExternalURL:(NSURL*)url inSafari:(BOOL)inSafari {
  // blah blah blah, previous logic, check `inSafari` at critical point (passing to internal methods if needed) and calling either `openURL:` or our own in-app stuff
}
```

**Comment by bgerstle** [comment]
:-1: should probably use wikipedia zero alert handling for the case when a user is on a W0 network so we can warn them before leaving (as is our policy)

**Comment by coreyfloyd** [comment]
@bgerstle ^

**Comment by bgerstle** [comment]
code looks good, just need to remember how to test Wikipedia 0 handling. To be continued...

**Comment by bgerstle** [comment]
tested wikipedia zero (with our old "faking" code) :+1: 

image

**Comment by montehurd** [comment]
:+1:

**Pull request merged**